File: backend/models/inventory_slot.py
"""
Inventory Slot model for inventory management
"""

from sqlalchemy import Column, Integer, String, DateTime, Date, Numeric, Boolean, CheckConstraint, text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from backend.database import Base


class InventorySlot(Base):
    """Inventory Slot model for tracking available inventory"""
    __tablename__ = "inventory_slots"

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()"), index=True)
    date = Column(Date, nullable=False, index=True)
    hour = Column(Integer, CheckConstraint("hour >= 0 AND hour <= 23"), nullable=False, index=True)
    break_position = Column(String(10))  # A, B, C, etc.
    daypart = Column(String(50))
    available = Column(Integer, default=3600)  # Available seconds in the hour
    booked = Column(Integer, default=0)  # Booked seconds
    sold_out = Column(Boolean, default=False, index=True)
    revenue = Column(Numeric(10, 2), default=0)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    def __repr__(self):
        return f"<InventorySlot(date='{self.date}', hour={self.hour}, available={self.available})>"

File: archive/librelog-web-api-original/src/main/java/com/onelpro/librelog/models/InventorySlot.java
package com.onelpro.librelog.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

/**
 * Entity representing an inventory slot in the system.
 */
@Entity
@Table(name = "inventory_slots", indexes = {
        @Index(name = "ix_inventory_slots_id", columnList = "id"),
        @Index(name = "ix_inventory_slots_date", columnList = "date"),
        @Index(name = "ix_inventory_slots_hour", columnList = "hour"),
        @Index(name = "ix_inventory_slots_sold_out", columnList = "sold_out")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class InventorySlot {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "date", nullable = false)
    private LocalDate date;

    @Column(name = "hour", nullable = false)
    private Integer hour;

    @Column(name = "break_position", length = 10)
    private String breakPosition;

    @Column(name = "daypart", length = 50)
    private String daypart;

    @Column(name = "available")
    @Builder.Default
    private Integer available = 3600;

    @Column(name = "booked")
    @Builder.Default
    private Integer booked = 0;

    @Column(name = "sold_out")
    @Builder.Default
    private Boolean soldOut = false;

    @Column(name = "revenue", precision = 10, scale = 2)
    @Builder.Default
    private BigDecimal revenue = BigDecimal.ZERO;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;
}

